model-optimizer/mo/utils/broadcasting.py

```diff
@@ -79,6 +79,23 @@ def bi_directional_shape_broadcasting(input_shape_1: np.array, input_shape_2: np
     return np.maximum(shape_1, shape_2)
 
 
+def explicit_shape_broadcasting(input_shape: np.array, target_shape: np.array, axes_mapping: np.array) -> np.array:
+    """
+    Explicit shape broadcasting of input tensor. Resulting shape is equal to target_shape except for axes specified in axes_mapping
+    :param input_value: input value to broadcast
+    :param target_shape: target shape
+    :param axes_mapping: a list of axis indices, each index maps an axis from the input_value to axis in the output
+    :return: broadcasted shape
+    """
+    assert np.all(np.diff(axes_mapping) >= 0), "axes_mapping is not sorted"
+
+    res = target_shape.copy()
+    for i, axis in enumerate(axes_mapping):
+        assert axis < len(res), "axis value from axes_mapping exceeds rank of target_shape"
+        res[axis] = input_shape[i]
```

Comment:
Please, check the nGraph implementation for the following case: input_shape = [1, 7], target_shape = [5,2,7,3] and axes_mapping = [1, 2]. I want to check whether broadcasting of 1 to N is allowed, or dimensions of the input tensor being mapped must be equal to the respective target dimensions. Based on the result of this check we may need to add an assert here that `res[axis] == input_shape[i]`

Reply:
In nGraph there is only one check: if mode is `explicit`' it only checks if the input with `axes_mapping` is present. So added value checks to MO.